docs/DESIGN.md

```diff
@@ -1,67 +1,79 @@
 
-  * Metrics collector that is deployed to each kafka broker
-  * Central doctorkafka service that analyzes broker status and execute kafka operation commands
+The following diagram shows the high level design:
 
-The following diagram shows the high level design. DoctorKafka is composed of two parts: i) the metrics collector that deploys on every kafka broker; 2) the central failure detection, workload balancing, and partition reassignment logic. The metric collectors send metrics to a kafka topic that the central DoctorKafka service read from. DoctorKafka takes actions and also log its action to another topic that can be viewed through web UI. DoctorKafka only takes confident actions, and send out alerts when it is not confident on taking actions. 
+<img src="images/high_level_design.png" width="800px"/>
 
-![doctorkafka diagram](doctorkafka_diagram.png)
-<img src="doctorkafka_diagram.png" width="160">
 
-
-#### Kafka Metrics collection
+## Kafka metrics collection agent (KafkaStats)
 
 DoctorKafka needs accurate kafka metrics to make sound decisions. As Kafka workload is mostly network bounded, DoctorKafka only uses topic partition network traffic metric to decide topic partition allocation. Currently kafka only have jmx metrics at topic level. It does not provide jmx metrics at replica level. Due to partition reassignment, etc., the traffic at topic level can vary a lot. Computing the normal network traffic of replicas becomes a challenge. 
 
 DoctorKafka deploys a metric collection agent on each kafka broker to collect metrics. The metric agent collect the following info for each broker: 
    *  Inbound and outbound network traffic for each leader replica
-   *  leader replicas on the broker
-   *  follower replicas on the broker
+   *  Leader replicas on the broker
+   *  Follower replicas on the broker
    *  In-reassignment topic replica
 
-Note that as of kafka 0.10.2, kafka only expose network traffic metrics for leader replicas. As follower replicas only have in-bound traffic, we can infer the follower replica traffic from leader replica traffic. 
+Note that as of kafka 0.10.2, Kafka only exposes network traffic metrics for leader replicas. As follower replicas only have in-bound traffic, we can infer the follower replica traffic from leader replica traffic. 
+
+
+## DoctorKafka Service
+
+The DoctorKafka service is a centralized manager that monitors Kafka clusters, detects failures and heals the system autonomously.
+
+
+### Plugin Framework
+In versions of DoctorKafka < 0.3.0, the operation logic of DoctorKafka was sitting in a single function that slowed down development iterations and prevented features to be delivered. We introduce the plugin framework in DoctorKafka 0.3 to provide a easier interface for developers to build new functionalities and users to fine tune their services. This framework is similar to a [control system](https://en.wikipedia.org/wiki/Control_system):
```

Comment:
How about rephrase  this paragraph as follows:
 
    Before 0.3.0, DoctorKafka implemented the cluster auto-healing logic in a single function. 
    That slowed down development velocity. The plugin framework